clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1469,7 +1470,25 @@ public static <K, V> Map<K, V> filterMap(final Map<K, V> map, final Predicate<En
      * @return a map including all elements in properties
      */
     public static Map<String, Object> propsToMap(Properties properties) {
-        return castToStringObjectMap(properties);
+        // This try catch block is to handle the case when the Properties object has non-String keys
+        // when calling the propertyNames() method. This is a workaround for the lack of a method that
+        // returns all properties including defaults and does not attempt to convert all keys to Strings.
+        Enumeration<?> enumeration;
+        try {
+            enumeration = properties.propertyNames();
+        } catch (ClassCastException e) {
+            throw new ConfigException("One or more keys is not a string.");
+        }
+        Map<String, Object> props = new HashMap<>();
```

Comment:
Pretty nit-picky, but I would name the map something other than `props`. `castToStringObjectMap()` uses `map`, FWIW.